### ISSUE TITLE
Events in overloaded operators

### ISSUE THREAD

**Issue by eshmoylova** [modelica/ModelicaSpecification#3239]
Can overloaded operators generate events? An operator record can overload behavior for binary operations, including the relational operations. The operators are implemented by functions. By default, functions do not generate events. So it would follow that overloaded operators cannot generate events. Is that intended? It seems odd that an expressions involving '>' and a potential discontinuity would not generate an event.

**Comment by HansOlsson** [comment]
I would agree that they cannot generate events.
Since I haven't seen any use of that overloading I don't know if it is a problem yet; and would just leave it as is for the moment.

In general special semantics for operators is a problem for overloading - C++ has a similar issue with overloaded `&&` and `||` (and `,` for more esoteric cases) and they are rarely overloaded.

**Comment by henrikt-ma** [comment]
I suppose `GenerateEvents = true` could still be used to get event-generating relations?

**Comment by HansOlsson** [comment]
> I suppose `GenerateEvents = true` could still be used to get event-generating relations?

Of course. So we are better off than C++ where overloaded && and || cannot short-circuit.
